profiles/views.py

```diff
@@ -37,6 +45,35 @@ class ProfileViewSet(RetrieveModelMixin, UpdateModelMixin, GenericViewSet):
     serializer_class_filled_out = ProfileFilledOutSerializer
     serializer_class_limited = ProfileLimitedSerializer
 
+    def subscribe_mailgun(self, email_optin, email):
+        """
+        it removes user from mailgun unsubscribes list.
+        https://documentation.mailgun.com/en/latest/api-suppressions.html#delete-a-single-unsubscribe
+
+        Args:
+            email_optin (bool): email optin flag
+            email (str): user email
+        """
+        if email_optin:
+            url = '{base}/unsubscribes/{email}'.format(base=settings.MAILGUN_URL, email=email)
+            response = requests.delete(url)
+            if response.status_code == statuses.HTTP_200_OK:
+                log.debug(
+                    "Removed user's email: %s from unsubscribes list. Message received: %s",
+                    email,
+                    response.text()
+                )
+
+    def update(self, request, *args, **kwargs):
+        response = super().update(request, *args, **kwargs)
+        self.subscribe_mailgun(request.data['email_optin'], request.data['email'])
```

Comment:
any suggestion if max limit of request reach or fail to remove user from mailgun unsub list?